src/arch/loongarch64/Makefile

```diff
@@ -0,0 +1,30 @@
+# Assembler section type character
+#
+ASM_TCHAR	:= @
+ASM_TCHAR_OPS	:= @
+
+# LoongArch64-specific flags
+#
+CFLAGS		+= -fstrength-reduce -fomit-frame-pointer
+CFLAGS		+= -falign-jumps=1 -falign-loops=1 -falign-functions=1
+CFLAGS		+= -fno-strict-aliasing
+
+# check whether _mno_explicit_relocs works
+ifeq ($(CCTYPE),gcc)
+RELOC_TEST = $(CC) -mno-explicit-relocs -x c -c /dev/null -o /dev/null >/dev/null 2>&1
+RELOC_FLAGS := $(shell $(RELOC_TEST) && $(ECHO) '-mno-explicit-relocs')
+WORKAROUND_CFLAGS += $(RELOC_FLAGS)
+endif
+
+# Debug
+NO_WERROR = 1
```

Comment:
If I remove `NO_WERROR`, I get a lot of printf errors, which I think needs to be fixed by change the common code.
```
core/memmap_settings.c:187:42: error: format '%llx' expects argument of type 'long long unsigned int', but argument 3 has type 'uint64_t' {aka 'long unsigned int'} [-Werror=format=]
  187 |                         DBGC ( settings, "MEMMAP %d start %08llx\n",
      |                                          ^~~~~~~~~~~~~~~~~~~~~~~~~~
  188 |                                i, region->start );
      |                                   ~~~~~~~~~~~~~
      |                                         |
      |                                         uint64_t {aka long unsigned int}
```

for example:
```
+++ b/src/include/compiler.h
@@ -449,6 +449,22 @@ char __debug_disable(OBJECT) = ( DBGLVL_MAX & ~DBGLVL_DFLT );
                }                                               \
        } while ( 0 )
 
+#if __SIZEOF_INT__ == 8
+# define    INT64_MODIFIER  ""
+# define    INT64_FORMAT    "i"
+# define    UINT64_FORMAT   "u"
+#elif __SIZEOF_LONG__ == 8
+# define    INT64_MODIFIER  "l"
+# define    INT64_FORMAT    "li"
+# define    UINT64_FORMAT   "lu"
+#elif __SIZEOF_LONG_LONG__ == 8
+# define    INT64_MODIFIER  "ll"
+# define    INT64_FORMAT    "lli"
+# define    UINT64_FORMAT   "llu"
+#else
+#error "Compiler provides no native 64-bit integer type"
+#endif
+
+++ b/src/core/memmap_settings.c
@@ -184,12 +184,12 @@ static int memmap_settings_fetch ( struct settings *settings,
                region = &memmap.regions[i];
                if ( include_start ) {
                        result += region->start;
-                       DBGC ( settings, "MEMMAP %d start %08llx\n",
+                       DBGC ( settings, "MEMMAP %d start %08"UINT64_FORMAT"x\n",
                               i, region->start );
                }
                if ( include_length ) {
                        result += ( region->end - region->start );
-                       DBGC ( settings, "MEMMAP %d length %08llx\n",
+                       DBGC ( settings, "MEMMAP %d length %08"UINT64_FORMAT"x\n",
                               i, ( region->end - region->start ) );
                }
        }

```

Reply:
There is zero chance of littering the codebase with `UINT64_FORMAT` or similar, and zero chance of allowing `NO_WERROR=1` as a default for any architecture.

A quick check shows that `sizeof(unsigned long)` and `sizeof(unsigned long long)` are both 64 bits for LoongArch64, which matches x86_64.  Change the definition of `uint64_t` in your `bits/stdint.h` and the problem should go away.

Reply:
> There is zero chance of littering the codebase with `UINT64_FORMAT` or similar, and zero chance of allowing `NO_WERROR=1` as a default for any architecture.

Patches that leave the format strings alone and instead cast the arguments to `( unsigned long long )` would be fine, but I think it will be easier for you to change your `uint64_t` definition since that's just a one-line change.

Reply:
Thank you, after changing `stdint.h`, the error is basically solved.
There is only one bug, which needs to be fixed by changing the define of UINT64 in `src/include/ipxe/efi/LoongArch64/ProcessorBind.h`.

Reply:
That `ProcessorBind.h` is imported from EDK2 along with all of the other UEFI specification headers, so you'd need to submit the fix to EDK2 first and we'll then pull the update from there.